free-1.md

```diff
@@ -10,34 +11,57 @@ _What need is met by your idea? Who are you helping? What is that person's
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
-
+As described above, the need is essentially to provide an interface that lets people test ideas in DOTA 2 quickly and easily. A GUI could work for this, but would effectively just be a GUI on top of a language that's doing all the backend computations. Thus, a first step is to make a DSL for it. Additionally, the language does not need to be turing complete, it can focus entirely on its domain. It might have batch testing and things that make use of more elaborate control flow, but these should all be in the backend. The goal is to make the interface as simple to use as possible. 
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I'm excited about this first and foremost because I play the game, and think it would be awesome to have something like this. I also know lots of people who play the game as well, and would love this. Second, it would be very helpful with respect to the constant debates about things in the game that are always hard to fact-check due to the difficulty of testing these things. I came up with it just by thinking of an interesting domain, and immediately thought of DOTA 2. 
 
 ### Domain
 _Describe the project's domain in five words._
 
+DOTA Hero/Item Balance Testing
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+The user would interact with this language just by writing scripts in either the shell, or a text editor. I'm imagining commands like:
+Hero {item1, item2, item3, item4, item5, item6} level 15;
+printing out the relevant stats for that hero. The core issue seems to be text or GUI as the core interaction. I claim that text is probably faster, as you could much mroe easily copy and paste the same 6 items into different heroes, rather than having to reselect everything every time. Text makes things like standard batch testing much easier than a GUI on its own.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+I sort of answered this with the section above, but the core idea is that you input some data:
+Hero (items) (level) (buffs/debuffs)
+and get as output some stats. You can then combine these things however you want (ie Hero1 damage > hero2 damage)
+or (at what level is Hero1 health > hero2 health). Finally, you could create a new hero, where you could input all of its base stats and abilities, and then treat it like a hero in the game already.
+
+Errors would generally just occur if the user inputs something that doesn't exist in the game (like giving a hero an item that doesn't exist).
 
```

Comment:
Ah, looks like you're already considering what I said above.  "(at what level is Hero1 health > hero2 health)"